feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -0,0 +1,223 @@
+import kotlinx.coroutines.CompletableDeferred
+
+private const val TAG = "PreviewScreen"
+
+// this is the preview surface display. It implements tap to focus, tap to zoom, and the
+@Composable
+fun PreviewDisplay(
+    viewModel: PreviewViewModel,
+    transformableState: TransformableState,
+    previewUiState: PreviewUiState,
+    deferredSurfaceProvider: CompletableDeferred<Preview.SurfaceProvider>
+) {
+    val onSurfaceProviderReady: (Preview.SurfaceProvider) -> Unit = {
+        Log.d(TAG, "onSurfaceProviderReady")
+        deferredSurfaceProvider.complete(it)
+    }
+    lateinit var viewInfo: View
+
+    BoxWithConstraints(
+        Modifier
+            .fillMaxSize()
+            .background(Color.Black)
+            .pointerInput(Unit) {
+                detectTapGestures(
+                    onDoubleTap = { offset ->
+                        // double tap to flip camera
+                        Log.d(TAG, "onDoubleTap $offset")
+                        viewModel.flipCamera()
+                    },
+                    onTap = { offset ->
+                        // tap to focus
+                        try {
+                            viewModel.tapToFocus(
+                                viewInfo.display,
+                                viewInfo.width,
+                                viewInfo.height,
+                                offset.x, offset.y
+                            )
+                            Log.d(TAG, "onTap $offset")
+                        } catch (e: UninitializedPropertyAccessException) {
+                            Log.d(TAG, "onTap $offset")
+                            e.printStackTrace()
+                        }
+                    }
+                )
+            },
+
+        contentAlignment = Alignment.Center
+    ) {
+        val maxAspectRatio: Float = maxWidth / maxHeight
+        val aspectRatio: Float =
+            previewUiState.currentCameraSettings.aspect_ratio.ratio.toFloat()
+        val shouldUseMaxWidth = maxAspectRatio <= aspectRatio
+        val width = if (shouldUseMaxWidth) maxWidth else maxHeight * aspectRatio
+        val height = if (!shouldUseMaxWidth) maxHeight else maxWidth / aspectRatio
+        Box(
+            modifier = Modifier
+                .width(width)
+                .height(height)
+                .transformable(state = transformableState)
+
+        ) {
+            CameraPreview(
+                modifier = Modifier
+                    .fillMaxSize(),
+                onSurfaceProviderReady = onSurfaceProviderReady,
+                onRequestBitmapReady = {
+                    val bitmap = it.invoke()
+                },
+                setSurfaceView = { s: View ->
+                    viewInfo = s
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun FlipCameraButton(
+    modifier: Modifier = Modifier,
+    enabledCondition: Boolean,
+    onClick: () -> Unit
+) {
+    Box(modifier = modifier) {
```

Comment:
the box acts as a spacer for the elements occupying the same row as the capture button.

Since the capture button is centered relative to this box, it will maintain the same relative position regardless of the device it is viewed on